schematic/store/synapse.py

```diff
@@ -1335,8 +1338,43 @@ def upload_manifest_file(
 
         return manifest_synapse_file_id
 
+    async def get_async_annotation(self, synapse_id: str) -> Dict[str, Any]:
+        """get annotations asynchronously
+
+        Args:
+            synapse_id (str): synapse id of the entity that the annotation belongs
+
+        Returns:
+            Dict[str, Any]: The requested entity bundle matching
+            <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/entitybundle/v2/EntityBundle.html>
+        """
+        return await get_entity_id_bundle2(
+            entity_id=synapse_id,
+            request={"includeAnnotations": True},
+            synapse_client=self.syn,
+        )
+
+    async def store_async_annotation(self, annotation_dict: dict) -> Annotations:
+        """store annotation in an async way
+
+        Args:
+            annotation_dict (dict): annotation in a dictionary format
+
+        Returns:
+            Annotations: The stored annotations.
+        """
+        annotation_data = Annotations.from_dict(
+            synapse_annotations=annotation_dict["annotations"]["annotations"]
+        )
+        annotation_class = Annotations(
+            annotations=annotation_data,
+            etag=annotation_dict["annotations"]["etag"],
+            id=annotation_dict["annotations"]["id"],
+        )
+        return await annotation_class.store_async(self.syn)
```

Comment:
```suggestion
        return await annotation_class.store_async(synapse_client=self.syn)
```

Personally, I really like using kw arguments. I am also considering a requirement that `synapse_client` is always a kwargument for all functions. What if your opinion on that change?

My thought: If I make a change to any of these functions my preference would be to order it before `synapse_client`. However, if I do not enforce it be referenced by kwarg, then it would be a breaking change to add an argument earlier in the list.

Further reading: https://realpython.com/python-asterisk-and-slash-special-parameters/#in-short-the-python-asterisk-and-slash-control-how-to-pass-values-to-functions

Reply:
Thanks for the comment Bryan! I personally really like keyword arguments as well. I think when reviewing the code, it is clearer what is being passed in what parameters for functions that require keyword arguments. 

hmm how many functions are you thinking of changing? I think probably it is not a good idea to change functions in `synapseclient.client` since that will introduce breaking changes that we would need to cope with immediately. But since schematic just started to utilize functions from `synapseclient.models`,  we have time to adapt :)

Reply:
> hmm how many functions are you thinking of changing? I think probably it is not a good idea to change functions in synapseclient.client since that will introduce breaking changes that we would need to cope with immediately. But since schematic just started to utilize functions from synapseclient.models, we have time to adapt :)

I would only be making changes in the new code. Everything else would remain as is to allow for backwards compatibility.